Mathlib/RepresentationTheory/GroupCohomology/LowDegree.lean

```diff
@@ -0,0 +1,328 @@
+  have h2 : _ ≫ ModuleCat.ofHom (dTwo A) = _ ≫ _ := congr_arg ModuleCat.ofHom (dTwo_comp_eq A)
+  simp only [←LinearEquiv.toModuleIso_hom] at h1 h2
+  simp only [(Iso.eq_inv_comp _).2 h2, (Iso.eq_inv_comp _).2 h1,
+    Category.assoc, Iso.hom_inv_id_assoc, HomologicalComplex.d_comp_d_assoc, zero_comp, comp_zero]
+
+end Differentials
+
+section Cocycles
+
+/-- The 1-cocycles `Z¹(G, A)` of `A : Rep k G`, defined as the kernel of the map
+`Fun(G, A) → Fun(G × G, A)` sending `(f, (g₁, g₂)) ↦ ρ_A(g₁)(f(g₂)) - f(g₁g₂) + f(g₁).` -/
+def oneCocycles : Submodule k (G → A) := LinearMap.ker (dOne A)
+
+/-- The 2-cocycles `Z²(G, A)` of `A : Rep k G`, defined as the kernel of the map
+`Fun(G × G, A) → Fun(G × G × G, A)` sending
+`(f, (g₁, g₂, g₃)) ↦ ρ_A(g₁)(f(g₂, g₃)) - f(g₁g₂, g₃) + f(g₁, g₂g₃) - f(g₁, g₂).` -/
+def twoCocycles : Submodule k (G × G → A) := LinearMap.ker (dTwo A)
+
+variable {A}
+
+theorem mem_oneCocycles_def (f : G → A) :
+    f ∈ oneCocycles A ↔ ∀ g : G × G, A.ρ g.1 (f g.2) - f (g.1 * g.2) + f g.1 = 0 :=
+  LinearMap.mem_ker.trans Function.funext_iff
+
+theorem mem_oneCocycles_iff (f : G → A) :
+    f ∈ oneCocycles A ↔ ∀ g : G × G, f (g.1 * g.2) = A.ρ g.1 (f g.2) + f g.1 := by
+  simp_rw [mem_oneCocycles_def, sub_add_eq_add_sub, sub_eq_zero, eq_comm]
+
+-- not sure whether preferable to the arguments `(f : G → A) (hf : f ∈ oneCocycles A)`
+theorem oneCocycles_map_one (f : oneCocycles A) : f.1 1 = 0 := by
+  have := (mem_oneCocycles_def f.1).1 f.2 (1, 1)
+  simpa only [map_one, LinearMap.one_apply, mul_one, sub_self, zero_add] using this
+
+theorem oneCocycles_map_inv (f : oneCocycles A) (g : G) :
+    A.ρ g (f.1 g⁻¹) = - f.1 g := by
+  rw [← add_eq_zero_iff_eq_neg, ← oneCocycles_map_one f, ← mul_inv_self g,
+    (mem_oneCocycles_iff f.1).1 f.2 (g, g⁻¹)]
+
+theorem mem_twoCocycles_def (f : G × G → A) :
+    f ∈ twoCocycles A ↔ ∀ g : G × G × G,
+      A.ρ g.1 (f (g.2.1, g.2.2)) - f (g.1 * g.2.1, g.2.2) +
+        f (g.1, g.2.1 * g.2.2) - f (g.1, g.2.1) = 0 :=
+  LinearMap.mem_ker.trans Function.funext_iff
+
+theorem mem_twoCocycles_iff (f : G × G → A) :
+    f ∈ twoCocycles A ↔ ∀ g : G × G × G,
+      f (g.1 * g.2.1, g.2.2) + f (g.1, g.2.1) =
+        A.ρ g.1 (f (g.2.1, g.2.2)) + f (g.1, g.2.1 * g.2.2) := by
+  simp_rw [mem_twoCocycles_def, sub_eq_zero, sub_add_eq_add_sub, sub_eq_iff_eq_add, eq_comm,
+    add_comm (f (Prod.fst _, _))]
+
+theorem twoCocycles_map_one_fst (f : twoCocycles A) (g : G) :
+    f.1 (1, g) = f.1 (1, 1) := by
+  have := ((mem_twoCocycles_iff f.1).1 f.2 (1, (1, g))).symm
+  simpa only [map_one, LinearMap.one_apply, one_mul, add_right_inj, this]
+
+-- should maybe be the other way around, with a different name?
+theorem twoCocycles_map_one_snd (f : twoCocycles A) (g : G) :
+    f.1 (g, 1) = A.ρ g (f.1 (1, 1)) := by
+  have := (mem_twoCocycles_iff f.1).1 f.2 (g, (1, 1))
+  simpa only [mul_one, add_left_inj, this]
+
+-- no idea for naming
```

Comment:
I think the statements and names are ok. You may remove the comments on lies 248/254.

bors d+